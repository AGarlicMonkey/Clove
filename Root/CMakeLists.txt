add_subdirectory(Definitions)
add_subdirectory(Delegate)
add_subdirectory(Event)
add_subdirectory(Expected)
add_subdirectory(Log)
add_subdirectory(Memory)