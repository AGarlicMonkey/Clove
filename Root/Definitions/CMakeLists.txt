#CMake
option(GAR_ENABLE_DEBUG_DEFINITION "Sets the GARLIC_DEBUG macro to 1. Allowing for various debug functionality / information to be enabled" OFF)

#Library
add_library(
	Root.Definitions INTERFACE
)

#Include
target_include_directories(
	Root.Definitions 
	
	INTERFACE 
		include
)

#Preprocessor
target_compile_definitions(
	Root.Definitions 

	INTERFACE
		#Platform
		GARLIC_PLATFORM_WINDOWS=$<PLATFORM_ID:Windows>
		GARLIC_PLATFORM_LINUX=$<PLATFORM_ID:Linux>
		GARLIC_PLATFORM_MACOS=$<PLATFORM_ID:Darwin>

		#Flags
		GARLIC_DEBUG=$<BOOL:${GAR_ENABLE_DEBUG_DEFINITION}>
		GARLIC_ENABLE_ASSERTIONS=$<BOOL:${GAR_ENABLE_ASSERTIONS}>
)