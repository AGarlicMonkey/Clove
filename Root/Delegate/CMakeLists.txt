set(INCLUDE ${CMAKE_CURRENT_SOURCE_DIR}/include/Root/Delegate)
set(SOURCE ${CMAKE_CURRENT_SOURCE_DIR}/source)

#Library
add_library(
    Root.Delegate STATIC
        ${INCLUDE}/DelegateHandle.hpp
	    ${SOURCE}/DelegateHandle.cpp
	    ${INCLUDE}/MultiCastDelegate.hpp
	    ${INCLUDE}/MultiCastDelegate.inl
	    ${INCLUDE}/SingleCastDelegate.hpp
	    ${INCLUDE}/SingleCastDelegate.inl
)

#Include
target_include_directories(
	Root.Delegate 
	
	PUBLIC
		include
)

if(GAR_BUILD_TESTS)
	add_subdirectory(tests)
endif()