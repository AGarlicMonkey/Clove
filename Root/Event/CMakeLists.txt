set(INCLUDE ${CMAKE_CURRENT_SOURCE_DIR}/include/Root/Event)
set(SOURCE ${CMAKE_CURRENT_SOURCE_DIR}/source)

#Library
add_library(
    Root.Event STATIC
        ${INCLUDE}/EventContainerBase.hpp
        ${SOURCE}/EventContainerBase.cpp
	    ${INCLUDE}/EventDispatcher.hpp
        ${INCLUDE}/EventDispatcher.inl
        ${SOURCE}/EventDispatcher.cpp
        ${INCLUDE}/EventHandle.hpp
        ${SOURCE}/EventHandle.cpp
	    ${INCLUDE}/EventManager.hpp
        ${INCLUDE}/EventManager.inl
        ${SOURCE}/EventManager.cpp
	    ${INCLUDE}/EventTypes.hpp
)

#Include
target_include_directories(
	Root.Event 
	
	PUBLIC
		include
)