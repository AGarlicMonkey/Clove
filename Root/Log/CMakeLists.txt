option(GAR_ENABLE_ASSERTIONS "Enables the GARLIC_ASSERT macro. Allowing the program to debug break if certain conditions fail." ON)

set(INCLUDE ${CMAKE_CURRENT_SOURCE_DIR}/include/Root/Log)
set(SOURCE ${CMAKE_CURRENT_SOURCE_DIR}/source)

add_subdirectory(external)

#Library
add_library(
    Root.Log STATIC
        ${INCLUDE}/Log.hpp
	    ${INCLUDE}/Log.inl
	    ${SOURCE}/Log.cpp
)

#Include
target_include_directories(
	Root.Log 
	
	PUBLIC
		include
)

#Libraries
target_link_libraries(
    Root.Log 

    PUBLIC
        spdlog

    PRIVATE
        Root.Definitions
)

#Preprocessor
target_compile_definitions(
	Root.Log 

	PUBLIC
		GARLIC_ENABLE_ASSERTIONS=$<BOOL:${GAR_ENABLE_ASSERTIONS}>
)