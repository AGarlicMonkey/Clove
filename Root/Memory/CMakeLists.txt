option(GAR_ENABLE_MEMORY_DEBUGGING "Enables certain memory debugging tools such as extra logs" OFF)

set(INCLUDE ${CMAKE_CURRENT_SOURCE_DIR}/include/Root/Memory)
set(SOURCE ${CMAKE_CURRENT_SOURCE_DIR}/source)

#Library
add_library(
    Root.Memory STATIC
        ${INCLUDE}/ListAllocator.hpp
		${SOURCE}/ListAllocator.cpp
		${INCLUDE}/PoolAllocator.hpp
		${INCLUDE}/PoolAllocator.inl
		${INCLUDE}/StackAllocator.hpp
		${SOURCE}/StackAllocator.cpp
)

#Include
target_include_directories(
	Root.Memory 
	
	PUBLIC
		include
)

#Libraries
target_link_libraries(
    Root.Memory 

	PUBLIC #Public because PoolAllocator.inl includes the headers
		Root.Definitions
		Root.Log
)

#Preprocessor
target_compile_definitions(
	Root.Memory

	#General
	PRIVATE
		GAR_ENABLE_MEMORY_DEBUGGING=$<BOOL:${GAR_CLV_ENABLE_MEMORY_DEBUGGING}>
)

if(GAR_BUILD_TESTS)
	add_subdirectory(tests)
endif()