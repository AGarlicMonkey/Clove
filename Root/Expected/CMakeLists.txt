set(INCLUDE ${CMAKE_CURRENT_SOURCE_DIR}/include/Root)

#Library
add_library(
    Root.Expected INTERFACE
)

#Include
target_include_directories(
	Root.Expected 
	
	INTERFACE 
		include
)

if(GAR_BUILD_TESTS)
    add_subdirectory(tests)
endif()